File: 백준/Silver/1789. 수들의 합/수들의 합.py
N = int(input())
n = 1
res = 0

while 1 :
    if N - (n*(n+1)/2) == 0:
        break
    elif  N - (n*(n+1)/2) < 0:
        n -= 1
        break
    
    n += 1
print(n)

File: 백준/Silver/1789. 수들의 합/수들의 합.ts
import * as readline from 'readline';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let S: number;

rl.on('line', (line: string) => {
  S = Number(line);
  rl.close();
});

rl.on('close', () => {
  console.log(solve(S))
  process.exit(0);
});

const solve = (s:number): number => {
  let sum = 0;
  let cnt = 0
  let i = 1;
  
  while (sum <= s){
    sum += i
    cnt++
    i++
  }

  return cnt-1
}